src/tests/resistance/handshake.rs

```diff
@@ -290,6 +331,141 @@ async fn r001_t4_HANDSHAKE_connect_as_field() {
+    assert!(run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Send an empty field.
+    let cfg = gen_cfg(String::new());
+    assert!(run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Use a huge value which the node will always reject.
+    let cfg = gen_cfg(gen_huge_string(WS_HTTP_HEADER_INVALID_SIZE));
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+}
+
+#[allow(non_snake_case)]
+#[tokio::test]
+async fn r001_t7_HANDSHAKE_network_time_field() {
+    // ZG-RESISTANCE-001
+    // Expected valid value for the "Network-Time" field in the handshake should be a valid number.
+
+    let debug = Debug::disable();
+
+    let gen_cfg = |time: String| SynthNodeCfg {
+        handshake: Some(HandshakeCfg {
+            http_network_time: Some(time),
+            ..Default::default()
+        }),
+        ..Default::default()
+    };
+
+    let time_now = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_secs()
+        - RIPPLE_EPOCH_OFFSET;
+    // Valid value for Network-Time
+    let cfg = gen_cfg(format!("{time_now}"));
+    assert!(run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Find the largest instance value which the node could accept.
+    // Send correct value but prepend lots of zeros.
+    let cfg = gen_cfg(format!("{:#07000}", time_now));
+    assert!(run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Invalid values for Network-Time
+    let cfg = gen_cfg(format!("{}", time_now - 24 * 60 * 60));
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    let cfg = gen_cfg(format!("{}", time_now + 24 * 60 * 60));
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    let cfg = gen_cfg(format!("{time_now}xyz"));
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Send an empty field.
+    let cfg = gen_cfg(String::new());
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Use a huge value which the node will always reject.
+    // Send correct value but prepend too many zeros.
+    let cfg = gen_cfg(format!("{:#08000}", time_now));
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+}
+
+#[allow(non_snake_case)]
+#[tokio::test]
+async fn r001_t8_HANDSHAKE_upgrade_req_field() {
+    // ZG-RESISTANCE-001
+    // Expected valid value for the "Upgrade" field in the handshake should be valid.
+
+    let debug = Debug::disable();
+
+    let gen_cfg = |version: String| SynthNodeCfg {
+        handshake: Some(HandshakeCfg {
+            http_upgrade_req: version,
+            ..Default::default()
+        }),
+        ..Default::default()
+    };
+
+    // Valid value for Upgrade
+    let cfg = gen_cfg("XRPL/2.2".to_owned());
+    assert!(run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Find the largest instance value which the node could accept.
+    let cfg = gen_cfg(format!("{:#7700}", "XRPL/2.2"));
```

Comment:
does this repeat the "XRPL/2.2" string 7700 times, or how does the string look like?

because  I was gonna suggest trying "XRPL/2.2,XRPL/2.2"